js/queen.js

```diff
@@ -98,7 +123,11 @@ export class Queen extends Piece{
          * Get row increasing and col decreasing moves
          */
         for (let i = this.rowPos + 1, y = this.colPos - 1; i < maxRows && y >= minCols; i++, y--) {
-            if (board[i][y] !== undefined && board[i][y].getPlayerId() == this.playerId) {
+            if (board[i][y] && board[i][y].getPlayerId() == this.playerId) {
```

Comment:
`===` should be used instead of `==` on line `126`.

Reply:
Updated.